opentelemetry-stdout/src/metrics/exporter.rs

```diff
@@ -76,57 +76,139 @@ impl PushMetricsExporter for MetricsExporter {
+            .iter()
+            .enumerate()
+            .for_each(|(index, kv)| {
+                if index == 0 {
+                    println!("\t\tScope Attributes:");
+                }
+                println!("\t\t\t ->  {}: {}", kv.key, kv.value);
+            });
+
+        metric.metrics.iter().enumerate().for_each(|(i, metric)| {
+            println!("Metric #{}", i);
+            println!("\t\tName         : {}", &metric.name);
+            println!("\t\tDescription  : {}", &metric.description);
+            println!("\t\tUnit         : {}", &metric.unit);
+
+            let data = metric.data.as_any();
+            if let Some(hist) = data.downcast_ref::<data::Histogram<i64>>() {
+                println!("\t\tType         : Histogram");
+                print_histogram(hist);
+            } else if let Some(hist) = data.downcast_ref::<data::Histogram<f64>>() {
+                println!("\t\tType         : Histogram");
+                print_histogram(hist);
+            } else if let Some(_hist) = data.downcast_ref::<data::ExponentialHistogram<i64>>() {
+                println!("\t\tType         : Exponential Histogram");
+                // TODO
+            } else if let Some(_hist) = data.downcast_ref::<data::ExponentialHistogram<f64>>() {
+                println!("\t\tType         : Exponential Histogram");
+                // TODO
+            } else if let Some(sum) = data.downcast_ref::<data::Sum<u64>>() {
+                println!("\t\tType         : Sum");
+                print_sum(sum);
+            } else if let Some(sum) = data.downcast_ref::<data::Sum<i64>>() {
+                println!("\t\tType         : Sum");
+                print_sum(sum);
+            } else if let Some(sum) = data.downcast_ref::<data::Sum<f64>>() {
+                println!("\t\tType         : Sum");
+                print_sum(sum);
+            } else if let Some(gauge) = data.downcast_ref::<data::Gauge<u64>>() {
+                println!("\t\tType         : Gauge");
+                print_gauge(gauge);
+            } else if let Some(gauge) = data.downcast_ref::<data::Gauge<i64>>() {
+                println!("\t\tType         : Gauge");
+                print_gauge(gauge);
+            } else if let Some(gauge) = data.downcast_ref::<data::Gauge<f64>>() {
+                println!("\t\tType         : Gauge");
+                print_gauge(gauge);
+            } else {
+                println!("Unsupported data type");
+            }
+        });
+    }
+}
+
+fn print_sum<T: std::fmt::Debug + DataPointValue>(sum: &data::Sum<T>) {
+    println!("\t\tSum DataPoints");
+    println!("\t\tMonotonic    : {}", sum.is_monotonic);
+    print_data_points(&sum.data_points);
+}
+
+fn print_gauge<T: std::fmt::Debug + DataPointValue>(gauge: &data::Gauge<T>) {
+    println!("\t\tGauge DataPoints");
+    print_data_points(&gauge.data_points);
+}
+
+fn print_histogram<T: std::fmt::Debug + DataPointValue>(histogram: &data::Histogram<T>) {
```

Comment:
nit: It would be less verbose to simply import `use std::fmt::Debug;` than to always write the fully qualified name.